### DISCUSSION TITLE
Path does not exist error in WSL 2

### DISCUSSION THREAD

**Discussion by chrisRidgers** [microsoft/winget-cli#972] [Q&A] [answered]
# Brief description of your issue

Attempting to validate a manifest file from inside WSL2 results in a 'path does not exist' error even though the path does exist.  I'm aware that authoring packages from WSL/ Linux probably wasn't intended, but I went to the effort of setting up SSH and such to clone the repository before realising the scripts to generate manifest files were PowerShell scripts and I carried on.

# Steps to reproduce

From inside a WSL instance:

Setup a function to run WSL as described here: https://github.com/microsoft/winget-cli/issues/234#issuecomment-647641718

```
$ winget validate --manifest /home/chris/Projects/Personal/winget-pkgs/manifests/w/WeakAuras/WeakAurasCompanion/3.3.1/WeakAuras.WeakAurasCompanion.yaml
'\\wsl$\Ubuntu\home\chris\Projects\Personal\winget-pkgs\manifests\w\WeakAuras\WeakAurasCompanion'
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
Path does not exist: /home/chris/Projects/Personal/winget-pkgs/manifests/w/WeakAuras/WeakAurasCompanion/3.3.1/WeakAuras.WeakAurasCompanion.yaml
$ pwd
/home/chris/Projects/Personal/winget-pkgs/manifests/w/WeakAuras/WeakAurasCompanion
$ ls 3.3.1
WeakAuras.WeakAurasCompanion.yaml
$ ls /home/chris/Projects/Personal/winget-pkgs/manifests/w/WeakAuras/WeakAurasCompanion/3.3.1/WeakAuras.WeakAurasCompanion.yaml
/home/chris/Projects/Personal/winget-pkgs/manifests/w/WeakAuras/WeakAurasCompanion/3.3.1/WeakAuras.WeakAurasCompanion.yaml
```
# Expected behavior

A notice regarding the correctness (or lack thereof) of my manifest file.

# Actual behavior

See output from terminal commands above.

# Environment

```
Windows Package Manager v0.3.11201 Preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19043.985
WSL2, ubuntu, zsh.
```

**Comment by denelon** [comment]
@chrisRidgers I'm not sure if this is the Windows Package Manager client, the terminal, or WSL 2. I'll see if I can have someone take a look.

**Comment by jedieaston** [comment]
~~What happens if you append `\\wsl$\<distroName>\` to the beginning of your path?~~

nevermind, I just reread the log file. winget would have to know to autocorrect that to a UNC path since cmd.exe can't.

**Comment by craigloewen-msft** [comment, marked as answer]
Could you try running this command instead:

`winget validate --manifest \\\\wsl$\\<yourDistroName>\\home\\chris\\Projects\\Personal\\winget-pkgs\\manifests\\w\\WeakAuras\\WeakAurasCompanion\\3.3.1\\WeakAuras.WeakAurasCompanion.yaml`

And replace `<yourDistroName>` with your distro name.

I was able to successfully use a similar command above on my machine.

**Comment by chrisRidgers** [comment]
```
$ winget va
lidate --manifest \\\\wsl$\\Ubuntu\\home\\chris\\Projects\\Personal\\winget-pkgs\\manifests\\w\\WeakAuras\\WeakAurasCompanion\\3.3.1\\WeakAuras.WeakAurasCompanion.yaml
'\\wsl$\Ubuntu\home\chris\Projects\Personal\winget-pkgs\manifests\w\WeakAuras\WeakAurasCompanion'
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
Manifest validation succeeded.
```

looks successful.

**Comment by denelon** [comment]
I'm going to convert this to discussion since it looks like this works :)
